packages/plugins/embarkjs/src/index.js

```diff
@@ -46,11 +57,62 @@ class EmbarkJS {
     this.events.setCommandHandler("embarkjs:contract:generate", this.addContractArtifact.bind(this));
     this.events.setCommandHandler("embarkjs:contract:runInVm", this.runInVm.bind(this));
 
+    if (!this.enabled) return;
+
     embark.registerActionForEvent("pipeline:generateAll:before", this.addEmbarkJSArtifact.bind(this));
     embark.registerActionForEvent("pipeline:generateAll:before", this.addContractIndexArtifact.bind(this));
+
+    if (this.config.blockchainConfig.enabled) {
+      this.setupEmbarkJS();
+    }
+
+    embark.registerActionForEvent("deployment:contract:deployed", {priority: 40}, this.registerInVm.bind(this));
+    embark.registerActionForEvent("deployment:contract:undeployed", this.registerInVm.bind(this));
+    embark.registerActionForEvent("deployment:contract:deployed", this.registerArtifact.bind(this));
+    embark.registerActionForEvent("deployment:contract:undeployed", this.registerArtifact.bind(this));
+  }
+
+
+
+  async setupEmbarkJS() {
+    if (!this.enabled) return;
+    this.events.on("blockchain:started", async () => {
+      await this.registerWeb3Object();
+      this.events.request("embarkjs:console:setProvider", 'blockchain', 'web3', '{web3}');
+    });
+    this.events.request("embarkjs:plugin:register", 'blockchain', 'web3', 'embarkjs-web3');
+    await this.events.request2("embarkjs:console:register", 'blockchain', 'web3', 'embarkjs-web3');
+  }
+
+  async registerWeb3Object() {
```

Comment:
I see that this code is both in the embarkjs module and the web3 one. I think this doesn't really belong here and we should only need the web3 one.

Reply:
Ok, I just checked the web3 plugin and I understand that we either use one plugin or the other. 
So IMO, the provider registration should be done in another module, since choosing EmbarkJS or Web3 doesn't change the fact that we need to register the provider.

Reply:
The right module to do this is in fact the web3 module. The reason that this is also being done here is because it maintains the behaviour of EmbarkJS for backwards compatibility and avoids some annoying race condition (koz the web3 object needs to be registered already in order to setup the embarkjs web3 provider)